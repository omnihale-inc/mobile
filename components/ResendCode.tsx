import { ReactNode } from "react";
import { Pressable, TextStyle, View, ViewStyle } from "react-native";

import { RegularText } from "./Texts";

const ResendCode = ({
  resend,
  onPress,
  styles,
  countDown
}: {
  resend: boolean;
  onPress: () => void;
  styles: { resend: ViewStyle; greenText: TextStyle };
  countDown: () => ReactNode;
}) => {
  return !resend ? (
    <View style={styles.resend}>
      <RegularText>Resend code in</RegularText>
      {countDown()}
    </View>
  ) : (
    <Pressable onPress={onPress}>
      <RegularText style={{ paddingTop: 12, ...styles.greenText }}>
        Resend code
      </RegularText>
    </Pressable>
  );
};

export default ResendCode;
