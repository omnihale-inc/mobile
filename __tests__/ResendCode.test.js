import renderer from "react-test-renderer";

import ResendCode from "../components/ResendCode";

it("render with counter", () => {
  const tree = renderer
    .create(
      <ResendCode
        resend={false}
        onPress={jest.fn()}
        countDown={() => "00:34"}
        styles={{ resend: {}, greenText: {} }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("render without counter", () => {
  const tree = renderer
    .create(
      <ResendCode
        resend
        onPress={jest.fn()}
        countDown={() => "00:34"}
        styles={{ resend: {}, greenText: {} }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
