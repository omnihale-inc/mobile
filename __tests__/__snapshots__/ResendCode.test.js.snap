// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with counter 1`] = `
<View
  style={{}}
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontFamily": "SatoshiRegular",
        },
      ]
    }
  >
    Resend code in
  </Text>
  00:34
</View>
`;

exports[`render without counter 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontFamily": "SatoshiRegular",
          "paddingTop": 12,
        },
      ]
    }
  >
    Resend code
  </Text>
</View>
`;
